### ISSUE TITLE
AnalyzeModel - VertiPaq Analyzer request timeout HTTP/408

### ISSUE THREAD

**Issue by cts14** [sql-bi/Bravo#517]
### Describe the problem

Error HTTP/408: Request timeout.

### Bravo version

1.0.0 (1.0.8339.28087)

### Tabular model

Power BI Desktop, import mode

### Power BI

SSAS 2017 LIVE Connection

### Steps to reproduce the behavior

1. Open PowerBI Desktop
2. Go to External Tools
3. Click Bravo
4. Error HTTP/408: Request timeout.

### Additional context

Note: I am able to connect to the same server using same credentials with DAX Studio (latest version)

### Screenshots

![BravoScreenshot](https://user-images.githubusercontent.com/56339801/199464944-7807c0ea-57c2-4ea1-bbba-aa1c6c10e080.png)

**Comment by albertospelta** [comment]
Could you share here the details of the `HTTP/500 Internal Server Error` error message?

![image](https://user-images.githubusercontent.com/1957788/199496760-926762df-076a-4047-90ee-d21a9984c988.png)

**Comment by albertospelta** [comment]
As far as I can see, the error `HTTP/500 Internal Server Error` no longer appears in the messages displayed.
Is the SSAS 2017 instance installed on a server in the local network or a virtual machine on cloud ?

**Comment by cts14** [comment]
Oh sorry I gave you details for the wrong event. We had to restart the SSAS service at that moment and that is why I got that message. The instance is installed in the local network. In case it helps when I check the running sessions while I am waiting from Bravo to fetch the model metadata I can see that it runs the following command: 

 `SELECT     DIMENSION_NAME AS TABLE_NAME,      TABLE_ID AS COLUMN_HIERARCHY_ID,     ROWS_COUNT - 3 AS COLUMN_CARDINALITY FROM $SYSTEM.DISCOVER_STORAGE_TABLES WHERE LEFT ( TABLE_ID, 2 ) = 'H$' ORDER BY TABLE_ID`

When I run this MDX query manually on SSMS I get a response within 50-60 seconds.

**Comment by cts14** [comment]
if I switch on the telemetry will that help you with more logs?

**Comment by albertospelta** [comment]
> if I switch on the telemetry will that help you with more logs?

Unfortunately not, this error does not generate a telemetry event.

> In case it helps when I check the running sessions while I am waiting from Bravo to fetch the model metadata I can see that it runs the following command:
> 
> SELECT DIMENSION_NAME AS TABLE_NAME, TABLE_ID AS COLUMN_HIERARCHY_ID, ROWS_COUNT - 3 AS COLUMN_CARDINALITY FROM $SYSTEM.DISCOVER_STORAGE_TABLES WHERE LEFT ( TABLE_ID, 2 ) = 'H$' ORDER BY TABLE_ID

The analysis of the model that Bravo performs requires the execution of multiple queries and the time required to complete all the queries exceeds 5 minutes which is the timeout that Bravo uses by default.

Would you mind running these queries via SSMS and let me know how long they take to complete?

**QUERY ColumnsData**
``` 
SELECT
    DIMENSION_NAME AS TABLE_NAME, 
    COLUMN_ID AS COLUMN_ID, 
    ATTRIBUTE_NAME AS COLUMN_NAME, 
    DATATYPE AS [Data Type],
    DICTIONARY_SIZE AS DICTIONARY_SIZE_BYTES,
    COLUMN_ENCODING AS COLUMN_ENCODING_INT
FROM  $SYSTEM.DISCOVER_STORAGE_TABLE_COLUMNS
WHERE COLUMN_TYPE = 'BASIC_DATA'
```

**QUERY ColumnsCardinality**
```
SELECT
    DIMENSION_NAME AS TABLE_NAME, 
    TABLE_ID AS COLUMN_HIERARCHY_ID,
    ROWS_COUNT - 3 AS COLUMN_CARDINALITY
FROM $SYSTEM.DISCOVER_STORAGE_TABLES
WHERE LEFT ( TABLE_ID, 2 ) = 'H$'
ORDER BY TABLE_ID
```

**QUERY ColumnsSegments**
```
SELECT 
    DIMENSION_NAME AS TABLE_NAME, 
    PARTITION_NAME, 
    COLUMN_ID AS COLUMN_NAME, 
    SEGMENT_NUMBER, 
    TABLE_PARTITION_NUMBER, 
    RECORDS_COUNT AS SEGMENT_ROWS,
    USED_SIZE,
    COMPRESSION_TYPE,
    BITS_COUNT,
    BOOKMARK_BITS_COUNT,
    VERTIPAQ_STATE {0}
FROM $SYSTEM.DISCOVER_STORAGE_TABLE_COLUMN_SEGMENTS
WHERE RIGHT ( LEFT ( TABLE_ID, 2 ), 1 ) <> '$'
```

**QUERY ColumnsHierarchies**
```
SELECT 
    DIMENSION_NAME AS TABLE_NAME, 
    COLUMN_ID AS STRUCTURE_NAME,
    SEGMENT_NUMBER, 
    TABLE_PARTITION_NUMBER, 
    USED_SIZE,
    TABLE_ID AS COLUMN_HIERARCHY_ID
FROM $SYSTEM.DISCOVER_STORAGE_TABLE_COLUMN_SEGMENTS
WHERE LEFT ( TABLE_ID, 2 ) = 'H$'
  AND SEGMENT_NUMBER = 0
```

**Comment by albertospelta** [comment]
> In case it helps...with DAX Studio* it takes 7 minutes to bring the VertiPaq Analyzer results.
Setting : Read statistics from data -> OFF

Would it be possible to test how long it takes to run if the option for reading statistics is turned on ?

**Comment by cts14** [comment]
I believe it crashes/fails to do so. That is why after researching online I found that if I switch it off it will run. I have exported the vpax from DaxStudio and imported into Bravo. It loaded succesfully. Note that our Model is 160GB

**Comment by albertospelta** [comment]
We have just pushed a fix that removes the 5-minute default timeout and, if you want to test, you can try the latest internal build. To upgrade Bravo to this version (`1.0.8342.18558`) you can change the update channel from `Stable` to `Dev` in the options panel as shown in the following image.

Please let me know if the timeout error still occurs even with this version. 
Thanks!

![image](https://user-images.githubusercontent.com/1957788/199706008-4705689e-19db-4117-bf0f-ffff3be625d2.png)

**Comment by cts14** [comment]
It worked. it took 7 minutes. Much appreciated

**Comment by albertospelta** [comment]
Fixed in version [1.0.1](https://github.com/sql-bi/Bravo/releases/tag/v1.0.1)
